src/Linux/Host/Broker/Impl/Services/LinuxSystemInfoService.cs

```diff
@@ -0,0 +1,172 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.Common.Core.IO;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.NetworkInformation;
+using System.Text;
+using System.Threading;
+
+namespace Microsoft.R.Host.Broker.Services {
+    public class LinuxSystemInfoService : ISystemInfoService {
+        private readonly IFileSystem _fs;
+
+        public LinuxSystemInfoService(IFileSystem fs) {
+            _fs = fs;
+        }
+
+        public double GetCpuLoad() {
+            var initial = GetCurrentCpuUsage();
+            Thread.Sleep(500);
+            var current = GetCurrentCpuUsage();
+
+            long initTimeSinceBoot = initial.user + initial.nice + initial.system + initial.idle + initial.iowait + initial.irq + initial.softirq + initial.steal;
+            long currentTimeSinceBoot = current.user + current.nice + current.system + current.idle + current.iowait + current.irq + current.softirq + current.steal;
+            long deltaTimeSinceBoot = currentTimeSinceBoot - initTimeSinceBoot;
+
+            long initIdleTime = initial.idle + initial.iowait;
+            long currentIdleTime = current.idle + current.iowait;
+            long deltaIdleTime = currentIdleTime - initIdleTime;
+
+            long deltaUsage = deltaTimeSinceBoot - deltaIdleTime;
+            return deltaUsage / deltaTimeSinceBoot;
+        }
+
+        public (long TotalVirtualMemory, long FreeVirtualMemory, long TotalPhysicalMemory, long FreePhysicalMemory) GetMemoryInformation() {
+            var meminfo = GetMemInfo();
```

Comment:
May need to handle `IOException` and `UnauthorizedAccessException` if `ReadAllLines` throws